lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -1187,6 +1187,23 @@ LLVMToSPIRVDbgTran::transDbgGlobalVariable(const DIGlobalVariable *GV) {
   if (DIDerivedType *StaticMember = GV->getStaticDataMemberDeclaration())
     Ops.push_back(transDbgEntry(StaticMember)->getId());
 
+  // Check if Ops[VariableIdx] has no information
+  if (Ops[VariableIdx] == getDebugInfoNoneId()) {
```

Comment:
Is it possible that we don't get `DebugInfoNone`, and also have DIExpression?

Reply:
In all my experiments I could not get Ops[VariableIdx]!=DebugInfoNone and also have a DIExpression, but I cannot guarantee this will not happen.  The DIExpression holds an initial value.  It seems to only be set if the variable is const, and thus the variable seems to be eliminated and Ops[VariableIdx]==DebugInfoNone.